src/main/java/seedu/address/logic/commands/gradetrackercommands/EditAssignmentCommand.java

```diff
@@ -1,16 +1,157 @@
+    private final ModuleName moduleName;
+    private final EditAssignmentDescriptor editAssignmentDescriptor;
+
+    /**
+     * @param index of the assignment in the gradetracker of the module
+     * @param editAssignmentDescriptor details to edit the assignment with
+     */
+    public EditAssignmentCommand(Index index, ModuleName moduleName,
+                                 EditAssignmentDescriptor editAssignmentDescriptor) {
+        requireNonNull(index);
+        requireNonNull(moduleName);
+        requireNonNull(editAssignmentDescriptor);
+
+        logger.info("Editing assignment at position " + index.toString() + " from:" + moduleName.toString() + "");
+        this.index = index;
+        this.moduleName = moduleName;
+        this.editAssignmentDescriptor = new EditAssignmentDescriptor(editAssignmentDescriptor);
+    }
+
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        return null;
+        requireNonNull(model);
+        List<Module> lastShownList = model.getFilteredModuleList();
+        Module moduleWithAssignment = null;
+
+        for (Module eachModule : lastShownList) {
+            if (eachModule.getName().equals(moduleName)) {
+                moduleWithAssignment = eachModule;
+                break;
+            }
+        }
+
+        if (moduleWithAssignment == null) {
+            throw new CommandException(MESSAGE_MODULE_INVALID);
+        }
+
+        if (index.getZeroBased() >= moduleWithAssignment.getGradeTracker().getAssignments().size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_ASSIGNMENT_DISPLAYED_INDEX);
+        }
+
+        Assignment assignmentToEdit = moduleWithAssignment.getGradeTracker().getSortedAssignments()
+                .get(index.getZeroBased());
+        Assignment editedAssignment = createEditedAssignment(assignmentToEdit, editAssignmentDescriptor);
+
+        if (!assignmentToEdit.isSameAssignment(editedAssignment)
+                && moduleWithAssignment.getGradeTracker().containsDuplicateAssignment(editedAssignment)) {
+            throw new CommandException(MESSAGE_DUPLICATE_ASSIGNMENT);
+        }
+
+        moduleWithAssignment.getGradeTracker().removeAssignment(assignmentToEdit);
+        moduleWithAssignment.getGradeTracker().addAssignment(editedAssignment);
+
+        logger.info("Assignment has been edited: " + assignmentToEdit.toString());
+
+        model.updateFilteredModuleList(PREDICATE_SHOW_ALL_MODULES);
+        return new CommandResult(String.format(MESSAGE_EDIT_ASSIGNMENT_SUCCESS, editedAssignment));
+    }
+
+    /**
+     * Creates and returns an {@code Assignment} with the details of {@code assignmentToEdit}
+     * edited with {@code editAssignmentDescriptor}.
+     */
+    private static Assignment createEditedAssignment(Assignment assignmentToEdit,
+                                                     EditAssignmentDescriptor editAssignmentDescriptor) {
+        assert assignmentToEdit != null;
```

Comment:
Perhaps you could consider adding assertion messages to be displayed. This can be helpful in understanding why the assertion was thrown.